sw/device/lib/dif/dif_rstmgr.h

```diff
@@ -24,6 +24,34 @@
 extern "C" {
 #endif  // __cplusplus
 
+/**
+ * Power on reset (POR).
+ */
+#define DIF_RSTMGR_REASON_POWERON (1 << 0)
+
+/**
+ * Low power exit (LOW_POWER_EXIT).
+ */
+#define DIF_RSTMGR_REASON_LOWPOWER_EXIT (1 << 1)
+
+/**
+ * Non-debug module (NDM).
+ */
+#define DIF_RSTMGR_REASON_NON_DEBUG_MODULE (1 << 2)
+
+/**
+ * Software issued request (SW).
+ */
+#define DIF_RSTMGR_REASON_SOFTWARE_REQUEST (1 << 3)
+
+/**
+ * Hardware requests (HW_REQ).
+ */
+#define DIF_RSTMGR_REASON_SYSRST_CTRL (1 << 4)
+#define DIF_RSTMGR_REASON_WATCHDOG (1 << 5)
+#define DIF_RSTMGR_REASON_POWER_UNSTABLE (1 << 6)
+#define DIF_RSTMGR_REASON_ESCALATION (1 << 7)
+
```

Comment:
As described in the documentation the `dif_rstmgr_reset_info_t` doesn't define the different Hardware reasons. I think that the best solution will be to split the `kDifRstmgrResetInfoHwReq`.

Reply:
I think that defining these defines here is not the right thing to do, otherwise they probably would have been named register bits.

I think maybe the better place would be `sw/device/lib/testing/rstmgr_testutils.<c,h>`.

@mundaym what do you think?

Reply:
You can have a look how this was approached on mask_rom
https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/lib/drivers/rstmgr.h;l=37

Reply:
I guess there are two things:
1) Whether we should define new variants
2) Eliminating redundancy

1) I guess it's fine to mimic the Mask ROM implementation
2) In my opinion, should either keep the defines or the existing `dif_rstmgr_reset_info_t`. Otherwise there are essentially two copies of the same thing - one more complete, the other one less complete. As `dif_rstmgr_reset_info_t` did exist before the defines, it seems reasonable to extend it. Alternatively, the DIF could match the HW exactly, and then have additional variants in `testutils`.